GamepadButtonAndAxisEvents/explainer.md

```diff
@@ -0,0 +1,216 @@
+
+rawgamepadinputchange event fires on the Gamepad object and delivers a snapshot of all gamepad input changes (axes and buttons) that occurred since the last event.
+
+- axesChanged and buttonsChanged contain the indices of all axes and buttons whose value changed since last input.
+
+- buttonsPressed and buttonsReleased include indices of buttons whose pressed state changed (i.e., transitioned to or from a pressed state) since last input.
+
+- The gamepadSnapshot provides the full current state of the gamepad, including button states, axis values, and metadata like ID, index, and timestamp.
+
+- A new rawgamepadinputchange event is dispatched for every individual input state change, enabling latency-sensitive applications (such as game streaming) to respond immediately to input events, without relying on polling. These events are not delayed or coalesced, and reflect each change as it happens, offering precise, real-time responsiveness.
+
+- This design is especially beneficial for scenarios like cloud gaming, rhythm games (guitar hero), or real-time multiplayer controls where timely input processing is critical.
+
+```js
+// Example rawgamepadinputchange Event:
+rawgamepadinputchange {
+  type: "rawgamepadchange",
+  gamepadSnapshot: Gamepad {
+    id: "Xbox Wireless Controller (STANDARD GAMEPAD Vendor: 045e Product: 02fd)",
+    index: 0,
+    connected: true,
+    mapping: "standard",
+    buttons: [
+      // index 0 - button A
+      { pressed: true, touched: false, value: 1.0 },
+      // index 1 - button B
+      { pressed: false, touched: true, value: 0.0 },
+      // index 2 - analog button
+      { pressed: false, touched: false, value: 0.5 },
+      ...
+    ],
+    // [left stick X, left stick Y, right stick X, right stick Y]
+    axes: [0.25, -0.5, 0.0, 0.0],
+    timestamp: 9123456.789
+  },
+  // Left stick X and Y moved since last event.
+  axesChanged: [0, 1],
+  // button index 0 was pressed, button index 2 value changed
+  buttonsValueChanged: [0, 2],
+  // button index 0 pressed
+  buttonsPressed: [0],
+  // button index 1 was touched
+  buttonsTouched: [1],
+  // High-res timestamp when change was detected
+  gamepadTimestamp: 9123456.789 
+}
+```
+## Proposed IDL
+```
+[Exposed=Window, SecureContext]
+interface Gamepad {
+  // New attributes
+  attribute EventHandler onrawgamepadinputchange;
+
+  // Existing attributes
+  readonly attribute DOMString id;
+  readonly attribute long index;
+  readonly attribute boolean connected;
+  readonly attribute DOMHighResTimeStamp timestamp;
+  readonly attribute GamepadMappingType mapping;
+  readonly attribute FrozenArray<double> axes;
+  readonly attribute FrozenArray<GamepadButton> buttons;
+};
+```
+### RawGamepadInputChangeEvent interface IDL, used for rawgamepadinputchange.
+```
+[Exposed=Window, SecureContent]
+interface RawGamepadInputChangeEvent : Event {
+  constructor(DOMString type, optional RawGamepadInputChangeEventInit eventInitDict = {});
```

Comment:
I just realized that you didn't add a definition for `RawGamepadInputChangeEventInit`. We should add it too.